src/components/Pill/PillButton/__tests__/PillButton.spec.tsx

```diff
@@ -8,52 +10,78 @@ describe('<PillButton> component', () => {
     const onClearMock = jest.fn();
     const name = 'Pill name';
     const content = 'This pill is in use';
+    const downArrowLabel = 'down arrow';
+    const upArrowLabel = 'up arrow';
+    const clearLabel = 'clear label';
 
-    let wrapper;
+    let view: RenderResult;
+
+    const getButtonByName = (inputName) => {
+        return screen.getByRole('button', { name: `${inputName}` });
+    };
 
     afterEach(() => {
         jest.resetAllMocks();
     });
 
     describe('in inactive, collapsed state (with minimal props)', () => {
         beforeEach(() => {
-            wrapper = mount(
-                <PillButton toggleDropdown={toggleDropdownMock} onClear={onClearMock} name={name} />
+            view = render(
+                <PillButton
+                    toggleDropdown={toggleDropdownMock}
+                    onClear={onClearMock}
+                    name={name}
+                    downArrowLabel={downArrowLabel}
+                    upArrowLabel={upArrowLabel}
+                    clearLabel={clearLabel}
+                />
             );
         });
 
         it('should render correctly', () => {
-            expect(toJson(wrapper)).toMatchSnapshot();
-            expect(wrapper.find('.PillButton__pill--isActive')).toHaveLength(0);
-            expect(wrapper.find('.PillButton__pill--isOpen')).toHaveLength(0);
+            expect(view.container).toMatchSnapshot();
+            expect(getButtonByName(downArrowLabel)).not.toHaveClass(
+                'PillButton__button PillButton__button--isOpen'
+            );
```

Comment:
I think this expect is misleading. you are checking for NOT having certain classes, but when you write it this way, it will pass even when there are some of these classes present and some not, which is the case here. Looking at the [docs](https://github.com/testing-library/jest-dom#tohaveclass), you can check for only the classes that you think they shouldn't be there, which I think is `PillButton__button--isOpen`. 

Having said that, maybe it makes sense to check that the down-arrow button is there, but e.g. not the clear button or the up-arrow one. I'm not sure the class is crucial here.

Reply:
:+1: